### PR TITLE
refactor: various minor refactors

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
various minor refactors, see commits

## What was done?
see commits

## How Has This Been Tested?
Building

## Breaking Changes
None

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: src/llmq/utils.cpp [hunk 1]
```diff
@@ -244,16 +244,16 @@ std::vector<std::vector<CDeterministicMNCPtr>> ComputeQuorumMembersByQuarterRota
...
-            quorumMembers[i].push_back(std::move(m));
+            quorumMembers[i].push_back(m);
         }
         for (const auto &m: previousQuarters.quarterHMinusC[i]) {
-            quorumMembers[i].push_back(std::move(m));
+            quorumMembers[i].push_back(m);
         }
         for (const auto &m: newQuarterMembers[i]) {
-            quorumMembers[i].push_back(std::move(m));
+            quorumMembers[i].push_back(m);
```
Comment (knst) [inline src/llmq/utils.cpp:256]: Why no need move here and why does it has been introduced?

Comment (PastaPastaPasta) [reply src/llmq/utils.cpp:256]: m is a const&; a const& cannot be cast to a && so this std::move does nothing but confuse

File: src/llmq/instantsend.cpp [hunk 2]
```diff
@@ -1211,8 +1210,7 @@ void CInstantSendManager::RemoveNonLockedTx(const uint256& txid, bool retryChild
 
     if (info.tx) {
         for (const auto& in : info.tx->vin) {
-            auto jt = nonLockedTxs.find(in.prevout.hash);
-            if (jt != nonLockedTxs.end()) {
+            if (auto jt = nonLockedTxs.find(in.prevout.hash); jt != nonLockedTxs.end()) {
```
Comment (knst) [inline src/llmq/instantsend.cpp:1213]: Nit: consider add const here

File: src/llmq/utils.cpp (hunk 1 above)
Comment (PastaPastaPasta) [reply src/llmq/utils.cpp:256]: I think this code would've been introduced by @ogabrielides; not sure why he used std::move :)

**Review by knst — COMMENTED** [review]
urACK

File: src/llmq/utils.cpp (hunk 1 above)
Comment (knst) [reply src/llmq/utils.cpp:256]: Maybe better to remove const then?

Comment (PastaPastaPasta) [reply src/llmq/utils.cpp:256]: not quite that simple but please see https://github.com/dashpay/dash/pull/5918

File: src/llmq/instantsend.cpp [hunk 3]
```diff
@@ -1255,11 +1253,9 @@ void CInstantSendManager::NotifyChainLock(const CBlockIndex* pindexChainLock)
...
 void CInstantSendManager::UpdatedBlockTip(const CBlockIndex* pindexNew)
 {
-    if (!fUpgradedDB) {
-        if (pindexNew->nHeight + 1 >= Params().GetConsensus().DIP0020Height) {
-            db.Upgrade(mempool);
-            fUpgradedDB = true;
-        }
+    if (!fUpgradedDB && pindexNew->nHeight + 1 >= Params().GetConsensus().DIP0020Height) {
+        db.Upgrade(mempool);
+        fUpgradedDB = true;
```
Comment (UdjinM6) [inline src/llmq/instantsend.cpp:1258]: I think it's safe to drop this `if` block, `fUpgradedDB` and `CInstantSendDb::Upgrade` completely actually.

File: src/llmq/instantsend.cpp [hunk 4]
```diff
@@ -1496,7 +1492,7 @@ void CInstantSendManager::AskNodesForLockedTx(const uint256& txid, const CConnma
 
 void CInstantSendManager::ProcessPendingRetryLockTxs()
 {
-    decltype(pendingRetryTxs) retryTxs = WITH_LOCK(cs_pendingRetry, return std::move(pendingRetryTxs));
+    auto retryTxs = WITH_LOCK(cs_pendingRetry, return pendingRetryTxs);
```
Comment (UdjinM6) [inline src/llmq/instantsend.cpp:1495]:
```suggestion
    const auto retryTxs = WITH_LOCK(cs_pendingRetry, return pendingRetryTxs);
```

File: src/llmq/utils.cpp (hunk 1 above)
Comment (UdjinM6) [reply src/llmq/utils.cpp:256]: should drop these changes from this PR then

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/llmq/utils.cpp (hunk 1 above)
Comment (PastaPastaPasta) [reply src/llmq/utils.cpp]: yup; done was just waiting to see if the other would gain traction.

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**